tests/test_classdecorators.py

```diff
@@ -0,0 +1,444 @@
+            DBOS.logger.info("I'm test_transaction")
+            return var2 + str(rows[0][0])
+
+        @classmethod
+        @dbos.communicator()
+        def test_communicator(cls, var: str) -> str:
+            DBOSTestClassClass.comm_counter += 1
+            DBOS.logger.info("I'm test_communicator")
+            return var
+
+    assert DBOSTestClassClass.test_workflow("bob", "bob") == "bob1bob"
+    wfh = dbos.start_workflow(DBOSTestClassClass.test_workflow, "bob", "bob")
+    assert wfh.get_result() == "bob1bob"
+    assert DBOSTestClassClass.txn_counter == 2
+    assert DBOSTestClassClass.wf_counter == 2
+    assert DBOSTestClassClass.comm_counter == 2
+
+
+def test_no_instname(dbos: DBOS) -> None:
+    @dbos.dbos_class()
+    class DBOSTestClassInst:
+        @dbos.workflow()
+        def test_workflow(self) -> str:
+            return "Nope"
+
+    with pytest.raises(Exception) as exc_info:
+        DBOSTestClassInst().test_workflow()
+    assert "Function target appears to be a class instance, but does not have `config_name` set"
+
+
+def test_simple_workflow_inst(dbos: DBOS) -> None:
+    @dbos.dbos_class()
+    class DBOSTestClassInst(DBOSConfiguredInstance):
+        def __init__(self) -> None:
+            super().__init__("bob", dbos)
+            self.txn_counter: int = 0
+            self.wf_counter: int = 0
+            self.comm_counter: int = 0
+
+        @dbos.workflow()
+        def test_workflow(self, var: str, var2: str) -> str:
+            self.wf_counter += 1
+            res = self.test_transaction(var2)
+            res2 = self.test_communicator(var)
+            DBOS.logger.info("I'm test_workflow")
+            return res + res2
+
+        @dbos.transaction()
+        def test_transaction(self, var2: str) -> str:
+            rows = DBOS.sql_session.execute(sa.text("SELECT 1")).fetchall()
+            self.txn_counter += 1
+            DBOS.logger.info("I'm test_transaction")
+            return var2 + str(rows[0][0])
+
+        @dbos.communicator()
+        def test_communicator(self, var: str) -> str:
+            self.comm_counter += 1
+            DBOS.logger.info("I'm test_communicator")
+            return var
+
+    inst = DBOSTestClassInst()
+    assert inst.test_workflow("bob", "bob") == "bob1bob"
+
+    wfh = dbos.start_workflow(inst.test_workflow, "bob", "bob")
+    stati = dbos.sys_db.get_workflow_status(wfh.get_workflow_uuid())
+    assert stati
+    assert stati["config_name"] == "bob"
+    assert stati["class_name"] == "DBOSTestClassInst"
+    stat = wfh.get_status()
+    assert stat
+    assert stat.config_name == "bob"
+    assert stat.class_name == "DBOSTestClassInst"
+
+    assert wfh.get_result() == "bob1bob"
+    assert inst.txn_counter == 2
+    assert inst.wf_counter == 2
+    assert inst.comm_counter == 2
+
+
+def test_forgotten_decorator(dbos: DBOS) -> None:
+    class DBOSTestRegErr(DBOSConfiguredInstance):
+        def __init__(self) -> None:
+            super().__init__("bob", dbos)
```

Comment:
I don't understand--doesn't this super().__init__ do the registration?

Reply:
It can, however I was making the rule that you put a @decorator at the class level.  The class-level decorator is the one that can actually see all the functions getting bound to the class and tie the whole thing together, so I think it makes sense to enforce the existence of one.  We can possibly relax the rule later.

Reply:
Maybe __init__ is a good chance to check that the class level decorator existed though...

Reply:
Yeah, that makes sense, it's a clear and easily enforceable rule.